data_steward/tools/retract_data_gcs.py

```diff
@@ -124,25 +135,38 @@ def retract(pids, bucket, found_files, folder_prefix, force_flag):
             response = get_response()
         if response == "Y":
             # Output and input file content initialization
-            retracted_file_string = StringIO()
-            input_file_string = gcs_utils.get_object(bucket, folder_prefix + file_name)
-            input_file_lines = input_file_string.split('\n')
+            retracted_file_string = BytesIO()
+            input_file_bytes = gcs_utils.get_object(bucket, folder_prefix + file_name, as_text=False)
+            input_file_lines = input_file_bytes.split(b'\n')
             input_header = input_file_lines[0]
             input_contents = input_file_lines[1:]
-            retracted_file_string.write(input_header + '\n')
+            retracted_file_string.write(input_header + b'\n')
             logger.info("Checking for person_ids %s in path %s" % (pids, file_gcs_path))
 
             # Check if file has person_id in first or second column
             for input_line in input_contents:
-                if input_line != '':
-                    cols = input_line.split(",")
-                    col_1 = cols[0]
-                    col_2 = cols[1]
-                    if (table_name in PID_IN_COL1 and get_integer(col_1) in pids) or \
-                            (table_name in PID_IN_COL2 and get_integer(col_2) in pids):
-                        lines_removed += 1
+                # ensure line is not empty
+                if input_line.strip() != b'':
+                    input_line = input_line.strip()
+                    cols = input_line.split(b',')
+                    # ensure at least two columns exist
+                    if len(cols) > 1:
+                        col_1 = cols[0]
+                        col_2 = cols[1]
+                        # skip if non-integer is encountered and keep the line as is
+                        try:
+                            if (table_name in PID_IN_COL1 and get_integer(col_1) in pids) or \
+                                    (table_name in PID_IN_COL2 and get_integer(col_2) in pids):
+                                lines_removed += 1
```

Comment:
nit:  add a comment here saying why the line isn't written back but the removed count is increased.

Reply:
Sure, I can add something like `do not write line back since it contains pid to retract`

Reply:
That would be great!  I know someone reading the code could eventually get there, but an explicit comment is so helpful.